cmd/kubectl.go

```diff
@@ -74,3 +116,37 @@ func runKubectl(ctx *cli.Context) error {
 	}
 	return nil
 }
+
+func writeKubeconfig(filePath string, kubeconfig []byte) error {
+	config, err := clientcmd.Load(kubeconfig)
+	if err != nil {
+		return err
+	}
+	return clientcmd.WriteToFile(*config, filePath)
+}
+
+func extractKubeconfigTokenID(kubeconfig api.Config) (string, error) {
+	if len(kubeconfig.AuthInfos) != 1 {
+		return "", fmt.Errorf("invalid kubeconfig, expected to contain exactly 1 user")
+	}
+	var parts []string
+	for _, val := range kubeconfig.AuthInfos {
```

Comment:
No need for the loop with the check for 1 above

Reply:
Spoke elsewhere, this is being used because the key is unknown.